src/overlays/actors/ovl_En_Reeba/z_en_reeba.c

```diff
@@ -21,45 +47,611 @@ const ActorInit En_Reeba_InitVars = {
+            }
+
+            this->unk_278 = 66;
+            this->actionfunc = func_80AE5E48;
+        } else {
+            this->unk_278 = 30;
+            this->actionfunc = func_80AE5A9C;
+        }
+    }
+}
+
+void func_80AE5A9C(EnReeba* this, GlobalContext* globalCtx) {
+    Vec3f randPos;
+    f32 scale;
+
+    if (this->unk_278 != 0) {
+        if ((this->unk_27E == 2) && ((this->unk_278 & 0xF) == 0)) {
+            randPos.x = this->actor.posRot.pos.x + Math_Rand_CenteredFloat(20.0f);
+            randPos.y = this->actor.posRot.pos.y + Math_Rand_CenteredFloat(20.0f);
+            randPos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(20.0f);
+            scale = 3.0f;
+
+            if (this->isBig) {
+                scale = 6.0f;
+            }
+
+            EffectSsEnIce_SpawnFlyingVec3f(globalCtx, &this->actor, &randPos, 150, 150, 150, 250, 235, 245, 255, scale);
+        }
+    } else {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIVA_DEAD);
+        func_80032C7C(globalCtx, &this->actor);
+        this->actionfunc = func_80AE5C38;
+    }
+}
+
+void func_80AE5BC4(EnReeba* this, GlobalContext* globalCtx) {
+    this->actor.speedXZ = -8.0f;
+    this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+    func_8003426C(&this->actor, 0x4000, 0xFF, 0, 8);
+    this->unk_278 = 14;
+    this->actor.flags &= ~1;
+    this->actionfunc = func_80AE5C38;
+}
+
+void func_80AE5C38(EnReeba* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+    Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    EnEncount1* spawner;
+
+    if (this->unk_278 != 0) {
+        if (this->actor.speedXZ < 0.0f) {
+            this->actor.speedXZ++;
+        }
+    } else {
+        this->actor.speedXZ = 0.0f;
+        Math_SmoothDownscaleMaxF(&this->scale, 0.1f, 0.01f);
+
+        if (this->scale < 0.01f) {
+            pos.x = this->actor.posRot.pos.x;
+            pos.y = this->actor.posRot.pos.y;
+            pos.z = this->actor.posRot.pos.z;
+            velocity.y = 4.0f;
+            EffectSsDeadDb_Spawn(globalCtx, &pos, &velocity, &accel, 120, 0, 255, 255, 255, 255, 255, 0, 0, 1, 9, true);
+
+            if (!this->isBig) {
+                Item_DropCollectibleRandom(globalCtx, &this->actor, &pos, 0xE0);
+            } else {
+                Item_DropCollectibleRandom(globalCtx, &this->actor, &pos, 0xC0);
+            }
+
+            if (this->actor.parent != NULL) {
+                spawner = (EnEncount1*)this->actor.parent;
+
+                if (spawner->actor.update != NULL) {
+                    if (!this->isBig) {
+                        if (spawner->numLeeversDead < 10) {
+                            spawner->numLeeversDead++;
+                        }
+                        // How many are dead?
+                        osSyncPrintf("\n\n");
+                        osSyncPrintf("\x1b[32m☆☆☆☆☆ 何匹ＤＥＡＤ？ ☆☆☆☆☆%d\n\x1b[m", spawner->numLeeversDead);
```

Comment:
VT macros (forgetting the basics Kappa)

Reply:
rofl I did 99% of this file months ago ok :mad: